README.md

```diff
@@ -6,66 +17,207 @@ A generic JavaScript helper library to query and manipulate Drupal 8 via core RE
 
 ## Setup
 
-* Run `npm i` to install development dependencies.
-* Run `npm t` to run tests and check coverage.
-* Run `npm run build` to create a browser version; located in `dist/waterwheel.js`.
+Install development dependencies.
+
+```
+npm i
+```
+
+Run tests and check coverage.
+
+```
+npm t
+```
+
+Build a browser version.
+
+```
+npm run build
+```
 
-## Usage
+## Documentation
 
-First, ensure that you have set up cross-origin resource sharing on your Drupal site to enable Waterwheel to perform necessary tasks.
+Ensure that you have set up cross-origin resource sharing on your Drupal site to enable Waterwheel to perform necessary tasks.
 
-From a server environment,
+### Require `waterwheel` in either a server or browser environment.
 
 ```javascript
+// Server
 const Waterwheel = require('waterwheel');
 const waterwheel = new Waterwheel('http://test.dev', {username: 'admin', 'password': '1234'});
 
-waterwheel.api.content.get(1)
+// Browser
+const waterwheel = new window.Waterwheel('http://test.dev', {username: 'admin', 'password': '1234'});
+```
+
+Waterwheel when instantiated accepts two arguments
+  - `base`: The base path for your Drupal instance. All request paths will be built from this base
+  - `credentials`: An object containing the `username` and `password` used to authenticate with Drupal.
+
+### Populate `waterwheel.api`
+
+```javascript
+waterwheel.populateResources()
   .then(res => {
-    // res
+    // An array of available resources
+  });
+```
+
+Waterwheel expects the [Waterwheel-Drupal module](https://www.drupal.org/project/waterwheel) to be installed and enabled. **This must be done before any subsequent api calls**.
```

Comment:
Should we use the waterwheel module to add a dependency on [entityqueryapi](https://www.drupal.org/project/entityqueryapi)? None of the query stuff will work without it.